defopt.py

```diff
@@ -118,8 +118,23 @@ def _create_parser(funcs, *args, **kwargs):
     return parser
 
 
-class _Formatter(RawTextHelpFormatter, ArgumentDefaultsHelpFormatter):
-    pass
+class _Formatter(RawTextHelpFormatter):
+
+    # modified from ArgumentDefaultsHelpFormatter to add type information, and
+    # remove defaults for varargs (`ZERO_OR_MORE`).
+    def _get_help_string(self, action):
+        info = []
+        if action.type is not None and '%(type)' not in action.help:
+            info.append('%(type)s')
+        if (action.default != argparse.SUPPRESS
+                and '%(default)' not in action.help
+                and action.option_strings
+                and action.nargs != argparse.ZERO_OR_MORE):
```

Comment:
This winds up eliminating more cases than I wanted. I specifically wanted to hide the default for variable positional arguments which have no user-defined default. This code also hides the default for keyword-argument lists, which *do* have a user-defined default, and should be displayed.

In retrospect I'm not sure why I was so opposed to displaying an empty list for the variable positional case. What do you think?